index.js

```diff
@@ -21,10 +27,8 @@ module.exports = (input, needle, replacement) => {
 		}
 
 		matchCount++;
-
 		const replaceStr = typeof replacement === 'string' ? replacement : replacement(needle, matchCount, input);
-
-		ret += input.slice(prevIndex, index) + replaceStr;
+		ret += (prevIndex === opts.fromIndex) ? input.slice(0, index) + replaceStr : input.slice(prevIndex, index) + replaceStr;
```

Comment:
I can try to make the code more understandable, let me try to explain. You are absolutely right, that ternary operator will only be true one time at the first iteration, but is there for a specific reason. 
In order to maintain a similar implementation logic to the one you had, it is necessary to have a way to catch the input behind the starting point. (if I was not clear enough, try to analyze the code with having in mind these two cases: 
```
-> ('foo bar', 'foo', 'bar')  => 'bar bar'
-> ('bar foo', 'foo', 'bar', 1)  => 'bar bar' (on this case, it is necessary to get the remaining 'b' behind the starting point.
```

I'll try to keep the code clear as soon as possible. But I'll appreciate another opinion regarding the previously stated  subject.